its/projects/MultiLanguageSupport/frontend/PageTwo.tsx

```diff
@@ -0,0 +1,83 @@
+import { cloneDeep } from 'lodash';
+import React, { useCallback, useEffect } from 'react';
+import useSelectState, { SelectState } from 'Helpers/Hooks/useSelectState';
+import ModelBase from './ModelBase';
+
+export type SelectContextAction =
+  | { type: 'reset' }
+  | { type: 'selectAll' }
+  | { type: 'unselectAll' }
+  | {
+      type: 'toggleSelected';
+      id: number;
+      isSelected: boolean;
+      shiftKey: boolean;
+    }
+  | {
+      type: 'removeItem';
+      id: number;
+    }
+  | {
+      type: 'updateItems';
+      items: ModelBase[];
+    };
+
+export type SelectDispatch = (action: SelectContextAction) => void;
+
+interface SelectProviderOptions<T extends ModelBase> {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  children: any;
+  items: Array<T>;
+}
+
+const SelectContext = React.createContext<
+  [SelectState, SelectDispatch] | undefined
+>(cloneDeep(undefined));
+
+export function SelectProvider<T extends ModelBase>(
+  props: SelectProviderOptions<T> // typescript:S6759
```

Comment:
I think we're only interested to see if javascript issues are raised. Could we simplify the page and have only the minimum required to raise 1 or 2 issues?